### ISSUE TITLE
About manipulation of gradients

### ISSUE THREAD

**Issue by kakusikun** [SCUT-AILab/DCP#5]
From [Clipping gradient by 10](https://github.com/SCUT-AILab/Discrimination-aware-Channel-Pruning-for-Deep-Neural-Networks/blob/master/dcp/channel_pruning.py#L736) and [nan correction](https://github.com/SCUT-AILab/Discrimination-aware-Channel-Pruning-for-Deep-Neural-Networks/blob/master/dcp/trainer.py#L339), you made some manipulation of gradients.

Why is the gradients clipped by 10, what's the effect do you find if the gradients is too large?
Why will the gradients be nan in segment wise training process, you encounter the nan first or you know the nan is inevitable then you correct it, which one?

**Comment by liujingcs** [comment]
We use gradient clipping and nan correction in another project. You can remove it safely. We set 10 in gradients clipping to prevent large gradients, which may cause Nan problem.

**Comment by yuanze-lin** [comment]
@JingLiu95 why pruned model's size is larger than pretrained model's size???

**Comment by liujingcs** [comment]
Do you have an example?

**Comment by yuanze-lin** [comment]
@JingLiu95 Sorry, I forgot to perform fine-tune.py after channel pruning.py
